Wander/app/src/main/java/com/teamwan/wander/Feedback.java

```diff
@@ -138,50 +146,103 @@ private void setTypefaces() {
+        }
+        LineDataSet dataSet = new LineDataSet(entries, "");
         dataSet.setColor(Color.RED);
-        LineData lineData = new LineData();
-        lineData.addDataSet(dataSet);
+        dataSet.setAxisDependency(YAxis.AxisDependency.LEFT);
+        LineData lineData = new LineData(labels, dataSet);
         return lineData;
     }
 
     /**
-     * todo: Creates bar chart from data and returns bitmap.
+     * Creates bar chart from data and returns BarData object.
      * @param type 1 if accuracy plot, 0 if response time plot
      */
-    public BarData createBarChart(int type){
-        return new BarData();
+    public BarData createBarChart(int type, float dataOn, float dataOff){
+        List<BarEntry> entries = new ArrayList<BarEntry>();
+        entries.add(new BarEntry(dataOff, 0));
+        entries.add(new BarEntry(dataOn, 1));
+        ArrayList<String> labels = new ArrayList<String>();
+        labels.add("Off Task");
+        labels.add("On Task");
+        BarDataSet dataSet = new BarDataSet(entries, "");
+        dataSet.setColor(Color.RED);
+        BarData barData = new BarData(labels, dataSet);
+        return barData;
     }
 
     /**
-     * todo: Fetches number of sessions that the player has completed
+     * Fetches number of sessions that the player has completed
      */
     public int fetchSessions(){
-        //int sessions = data.getTotalGames();
-        int sessions = 24;
-        //get total number of sessions here
-        return sessions;
+       return data.getNumberOfSessions();
     }
 
-    public double fetchResponse(){
-        //double response = data.getLatestART();
-        double response = 1.25;
-        //get avg. response time for the session here
+    public float fetchResponse(){
+        float response = data.getLatestNAverageResponses(1).get(0);
         return response;
     }
 
-    public int fetchAccuracy(){
-        //int accuracy = (int)data.getLatestCTP();
-        int accuracy = 95;
-        //get tap accuracy here
+    public float fetchAccuracy(){
+        float accuracy = data.getLastNGameSessions(1).get(0).getPercentage();
         return accuracy;
     }
+
+    public List<Float> fetchLineDataAcc(){
+        return data.getLatestNTaskCorrectness(6);
+    }
+
+    public List<Float> fetchLineDataTime(){
+        return data.getLatestNAverageResponses(6);
+    }
+
+    public float fetchBarDataTime(int task){
+        ArrayList<Float> values;
+        float avg = 0;
+        if (task==0){
+            values = data.getOnTaskCorrectnesses();
+            for (float v: values) {
+                avg += v;
+            }
+            return avg/values.size();
+        } else {
+            values = data.getOffTaskCorrectnesses();
+            for (float v: values) {
+                avg += v;
+            }
+            return avg/values.size();
+        }
+    }
+
+    public float fetchBarDataAcc(int task){
+        ArrayList<Float> values;
```

Comment:
This functionality is duplicate with the previous function. The only difference is that it uses a different set of values